File: src/main/java/com/jetbrains/snakecharm/lang/psi/elementTypes/SmkStubElementTypes.java
package com.jetbrains.snakecharm.lang.psi.elementTypes;

import com.intellij.psi.stubs.IStubElementType;
import com.jetbrains.snakecharm.lang.psi.*;
import com.jetbrains.snakecharm.lang.psi.impl.stubs.*;
import com.jetbrains.snakecharm.lang.psi.stubs.*;

public interface SmkStubElementTypes {
    IStubElementType<SmkRuleStub, SmkRule>  RULE_DECLARATION_STATEMENT = new SmkRuleElementType();
    IStubElementType<SmkCheckpointStub, SmkCheckPoint> CHECKPOINT_DECLARATION_STATEMENT = new SmkCheckpointElementType();
    IStubElementType<SmkSubworkflowStub, SmkSubworkflow> SUBWORKFLOW_DECLARATION_STATEMENT = new SmkSubworkflowElementType();
    IStubElementType<SmkModuleStub, SmkModule> MODULE_DECLARATION_STATEMENT = new SmkModuleElementType();
    IStubElementType<SmkUseStub, SmkUse> USE_DECLARATION_STATEMENT = new SmkUseElementType();
}

File: src/main/kotlin/com/jetbrains/snakecharm/lang/psi/elementTypes/SmkStubElementTypes.kt
package com.jetbrains.snakecharm.lang.psi.elementTypes

import com.intellij.psi.stubs.IStubElementType
import com.jetbrains.snakecharm.lang.psi.SMKCheckPoint
import com.jetbrains.snakecharm.lang.psi.SMKRule
import com.jetbrains.snakecharm.lang.psi.SmkSubworkflow
import com.jetbrains.snakecharm.lang.psi.impl.stubs.SmkCheckpointElementType
import com.jetbrains.snakecharm.lang.psi.impl.stubs.SmkRuleElementType
import com.jetbrains.snakecharm.lang.psi.impl.stubs.SmkSubworkflowElementType
import com.jetbrains.snakecharm.lang.psi.stubs.SmkCheckpointStub
import com.jetbrains.snakecharm.lang.psi.stubs.SmkRuleStub
import com.jetbrains.snakecharm.lang.psi.stubs.SmkSubworkflowStub

object SmkStubElementTypes {
    val RULE_DECLARATION: IStubElementType<SmkRuleStub, SMKRule> = SmkRuleElementType()
    val CHECKPOINT_DECLARATION: IStubElementType<SmkCheckpointStub, SMKCheckPoint> = SmkCheckpointElementType()
     val SUBWORKFLOW_DECLARATION: IStubElementType<SmkSubworkflowStub, SmkSubworkflow> = SmkSubworkflowElementType()
}